connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala

```diff
@@ -193,7 +197,11 @@ class MsSqlServerIntegrationSuite extends DockerJDBCIntegrationSuite {
         assert(types(10).equals("class java.math.BigDecimal"))
         assert(types(11).equals("class java.math.BigDecimal"))
         assert(row.getBoolean(0) == false)
-        assert(row.getInt(1) == 255)
+        if (flag) {
+          assert(row.getInt(1) == 255)
+        } else {
+          assert(row.getShort(1) == 255)
```

Comment:
LGTM later.
Surprisingly, SQL Server's tinyint does not support negative numbers.